Feat/adding home view

heliomar-pena opened the pull request:
## Explanation

I have added the home page, including the stories for the storybook where you can play with the different variables that receive the view and test distinct cases

I have to integrate it with the database, except by the user that is always a mock data, since the login is not working yet to get and doesn't exist some like a user "session".

## Screenshots

![LastRecord](https://user-images.githubusercontent.com/66505715/172985084-e75dfe7d-22eb-4091-8a31-316d8d36eb25.gif)

## How to test

1. Go to the netlify deploy link
2. **Enjoy testing the home!**

Storybook:
1. Go to the **onrender** deploy link
2. Search the Home page for the different components
3. Test the different components that are shown in Storybook
4. Let me know if you found any bugs on any component!  

## Story:

- https://github.com/Azordev/did-app/issues/14

heliomar-pena committed:
Added query to get user by Id and updated login query

heliomar-pena committed:
Added initial files to work on Home view

heliomar-pena committed:
Added text on HomeHeader

heliomar-pena committed:
Added some styles to header

heliomar-pena committed:
Added styles for header

heliomar-pena committed:
Fixing bug visual

heliomar-pena committed:
Added events query

heliomar-pena committed:
Merge branch 'dev' into feat/adding-home-view

heliomar-pena committed:
Added HomeHeader to Storybook and fixed some bugs

heliomar-pena committed:
Created EventCard component

heliomar-pena committed:
Added BookMark component

heliomar-pena committed:
Added convocatory label

heliomar-pena committed:
Added Horizontal Scroll Component

heliomar-pena committed:
Added Events list to the APP

heliomar-pena committed:
Some styles fixes

heliomar-pena committed:
Changed script for setup

heliomar-pena committed:
Added provider card

heliomar-pena committed:
Fixed Providers Query and remove Category

heliomar-pena committed:
Added Provider list to home list stories

heliomar-pena committed:
Added HomeLayout to storybook

heliomar-pena committed:
Refactored home code

heliomar-pena committed:
Added home layout styles

heliomar-pena committed:
Added events integration with the database

heliomar-pena committed:
Updated home queries

heliomar-pena committed:
Added integration with providers

heliomar-pena committed:
Some improves

heliomar-pena committed:
Fixing typo

heliomar-pena committed:
Fixing typescript error

heliomar-pena committed:
Linking provider with provider detail

heliomar-pena committed:
Linking cards to details

Israel-Laguan commented on src/components/ProviderCard/ProviderCard.stories.js:
```diff
@@ -0,0 +1,24 @@
+};
+
+// More on component templates: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args
+const Template = (args) => ({
```
What about naming it with something related to the component?

heliomar-pena replied:
"Template" is not visible for users on storybook, since I'm not exporting it. It's just used to created multiple examples with different props, as you can see on "Default"

```javascript
export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/vue/writing-stories/args
Default.args = {
  image_url: DIDLogo,
  id: '1',
};
```

The idea is that I can create another example without add all the settings of Template:

```javascript
export const CasaRosello = Template.bind({});
// More on args: https://storybook.js.org/docs/vue/writing-stories/args
Default.args = {
  image_url: CasaRosello,
  id: '3',
};
```

Then the user will see two options:

- Default
- CasaRosello

---

But I think it's a good theme of conversation, i.e, in this case I have just one example on Storybook, do you think if it's just one example on storybook is better to add the story directly instead of create a Template?

Israel-Laguan replied:
What about "ProviderCardTemplate" or similar for the template?

heliomar-pena replied:
Hmmm ok I think it's ok for the template, maybe it helps the developer to find quickly the story for that component

Israel-Laguan commented on src/components/ProviderCard/ProviderCard.stories.js:
```diff
@@ -0,0 +1,24 @@
+    return { args };
+  },
+  template:
+    '<div style="justify-content: center; background-color: #ccc; display: flex; padding: 10px;"><provider-card v-bind="args" /></div>',
+});
```
Can you move this style to a CSS or at least to a "style={{}}" kind of object, or maybe a string?

I am kinda nervous to see it hardcoded

heliomar-pena replied:
Of course! Will do

Israel-Laguan commented on src/services/queries/Users.js:
```diff
@@ -15,17 +16,21 @@ export const getAllUsersQuery = gql`
+  }
+`;
+
+export const getUserByIdQuery = gql`
```
Not sure if there is a standard for naming queries? @e-azocar do you know?

e-azocar replied:
I found this https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions

Israel-Laguan commented on src/services/queries/Users.js:
```diff
@@ -15,17 +16,21 @@ export const getAllUsersQuery = gql`
+
+export const getUserByIdQuery = gql`
+  ${userInfo}
+  query getUsers($id: uuid = "") {
```
I think this default is not valid, please check

Israel-Laguan commented on src/services/fragments/User.js:
```diff
@@ -0,0 +1,15 @@
+        expiration
+      }
+    }
+    username
```
please move this to the top

Israel-Laguan commented on src/pages/Home/utils/handleUserData.ts:
```diff
@@ -0,0 +1,55 @@
+  };
+
+  const parseUserData = (user: UserType) => ({
+    ...getExpirationDate(
+      user.member_information[0].subscriptions[0].expiration
+    ),
+    ...getUserName(user.member_information[0]),
+    ...getUserMemberCode(user.username),
```
Please notice this field don't exists anymore

https://did-db.herokuapp.com/console/data/did-db/schema/public/tables/users/relationships

You can use
```suggestion
    ...getExpirationDate(
      user.member_info.subscriptions[0].expiration
    ),
    ...getUserName(user.member_info),
    ...getUserMemberCode(user.username),
```

heliomar-pena replied:
Thanks for the correction, as it's mock data the changes wasn't reflected, I will update it

Israel-Laguan commented on src/services/fragments/User.js:
```diff
@@ -0,0 +1,15 @@
+export const userInfo = gql`
+  fragment userInfo on users {
+    avatar_url
+    member_information {
```
Please change to 
```suggestion
    member_info {
```

heliomar-pena replied:
Thanks!

Israel-Laguan commented on src/pages/Home/utils/handleUserData.ts:
```diff
@@ -0,0 +1,55 @@
+    };
+  };
+
+  const getUserMemberCode = (username: string) => {
+    const userCode = ref<string>(username);
+
+    return { userCode };
+  };
```
I don't get the purpose of this one, please explain

Seems like you are setting the state? Because if it's the case, it would change from 'VP000001' to "Victor Colmenarez", is that the intended behavior?

heliomar-pena replied:
Yes, it's setting the state, I did it that way just to avoid do it on parseUserData function (please notice that parseUserData is just calling functions and returning the results directly), but can be set on parseUserData if you think it's more readable

Israel-Laguan replied:
Yes, please use parseUserData instead

Israel-Laguan replied:
also notice the comment

> it would change from 'VP000001' to "Victor Colmenarez", is that the intended behavior?

heliomar-pena replied:
@Israel-Laguan It should save the user code, which is 'VP00000001'. Please let me know if that was your question

Israel-Laguan commented on src/pages/Home/utils/handleUserData.ts:
```diff
@@ -0,0 +1,55 @@
+  };
+
+  const parseUserData = (user: UserType) => ({
+    ...getExpirationDate(
+      user.member_information[0].subscriptions[0].expiration
+    ),
+    ...getUserName(user.member_information[0]),
```
```suggestion
    ...getExpirationDate(
      user.member_info.subscriptions[0].expiration
    ),
    ...getUserName(user.member_info),
```

Israel-Laguan commented on src/pages/Home/mock.ts:
```diff
@@ -0,0 +1,37 @@
+export type UserType = {
+  __typename: 'users';
+  avatar_url: string | null;
+  username: string;
```
Please notice this field is not used anymore

https://did-db.herokuapp.com/console/data/did-db/schema/public/tables/users/modify

You can use `member_code`

heliomar-pena replied:
Oh, thanks for notifying me !

e-azocar commented on src/actions/index.js:
```diff
@@ -1,3 +1,4 @@
 export * from './UserActions';
 export * from './ProvidersActions';
 export * from './ProductsActions';
+export * from './EventsActions';
```
```suggestion
export * from './EventsActions';

```
Please add EOF line

heliomar-pena replied:
I see it on my code editor 

![image](https://user-images.githubusercontent.com/66505715/173159440-288e0697-d927-4db2-87bf-e4f72ff87c0d.png)

I don't know why GitHub is not showing it :( But I think GitHub shows an EOF error when we need to add it

Israel-Laguan replied:
I remember there were 2 settings, one in git configuration (somethink like "clrf") and other in vscode (like addEndOfLine sorry I am from phone). Please check if you can add those on this project.

e-azocar commented on src/components/BookMark/BookMark.stories.js:
```diff
@@ -0,0 +1,21 @@
+// More on args: https://storybook.js.org/docs/vue/writing-stories/args
+Default.args = {
+  label: 'Convocatoria',
+};
```
same here

heliomar-pena replied:
I see it on my code editor, maybe github is trim() it
![image](https://user-images.githubusercontent.com/66505715/173159456-e24df01e-45ff-4348-a247-b77abd374d4d.png)

e-azocar commented on src/components/BookMark/BookMark.vue:
```diff
@@ -0,0 +1,18 @@
+    },
+  },
+});
+</script>
```
same here

heliomar-pena replied:
Can you check if on your code editor it exists? on my code editor it does :

![image](https://user-images.githubusercontent.com/66505715/173159492-d624aa45-9755-4e0e-9633-fe35a76236ef.png)

But github is not showing it

e-azocar commented:
@victor-heliomar I don't know if these items have image
![image](https://user-images.githubusercontent.com/61360270/173086225-a1ca4d1b-1235-4849-8098-0deb57a06c7b.png)

Israel-Laguan commented:
No they don't, as we don't have them online

Israel-Laguan commented:
Daniel gave me the images in png but we didn't set FileStack or similar yet

heliomar-pena commented:
@Israel-Laguan Yesterday I added two events yesterday for test, you will add official events? If it's so, please remove what I have created.

Too I have added images for the providers that were on the database, and changed the state, I remember that Rollux was inactive (is_active = false)

If u want I can remove these changes to left just the original data?

Israel-Laguan commented:
Sure, no problem

heliomar-pena committed:
Refactoring stories

heliomar-pena committed:
Added styles to a object

heliomar-pena committed:
Some code changes

heliomar-pena committed:
Refactoring queries

heliomar-pena committed:
using new fields on user query

heliomar-pena committed:
Changed member information for member info

heliomar-pena committed:
changed username for member_code

heliomar-pena committed:
Some style improves

heliomar-pena committed:
improving the conversion on queries

heliomar-pena committed:
Added disabled state to btn when is not membership active

Israel-Laguan commented on src/services/queries/Users.js:
```diff
@@ -14,18 +15,25 @@ export const getAllUsersQuery = gql`
+export const USER_BY_ID_QUERY = gql`
+  ${userInfo}
+  query getUsers($id: uuid!) {
+    users(where: { is_active: { _eq: true }, id: { _eq: $id } }) {
```
You can try to use user_by_pk to avoid bringing an array

heliomar-pena replied:
Of course, thanks !

Israel-Laguan commented on src/services/queries/Users.js:
```diff
@@ -1,9 +1,10 @@
+import { userInfo } from '../fragments';
 
-export const getAllUsersQuery = gql`
+export const USERS_QUERY = gql`
```
Is this query of use in the app? Like, you only need the user by member_code/password (USER_LOGIN_QUERY) maybe?

heliomar-pena replied:
No, it's not used on the APP, I have use it to test the integration with the GraphQL, but I should remove it hehe. Thanks for notifying me

Israel-Laguan commented on src/utils/apollo.types.ts:
```diff
@@ -109,18 +109,33 @@ export interface product {
+  title: string;
+  image_url?: string;
+  date: string;
+  type: 'PRIVATE' | 'PUBLIC';
```
I am thinking in change types to 

`ATTENDANCE | WORKSHOP`

- ATTENDANCE: It's a private event that a member has to inscribe to.
- WORKSHOP (maybe ANNOUNCE? I think @anaceciliamolinares should help here hehe): It's a public event with some requisites to participate but you have to send a form and/or follow instructions.

What do you think?

e-azocar replied:
I like `ATTENDANCE`, but I don't know if `WORKSHOP` is the best word

heliomar-pena replied:
I like both. 

"workshop" is weird to me since I didn't know the word, (sounds like "tienda de trabajo"?) but I've checked and it's equivalent to "Taller" in Spanish, remember me when I meet at the school with a group of people to learn/do a new thing between all.

If exist a better word could be considered

heliomar-pena replied:
BTW: I think I should change this:

```
   type: 'PRIVATE' | 'PUBLIC'; 
```

for a "enum" or something similar

anaceciliamolinares replied:
I think "workshop" is a more specific and catchy term that refers to developing your skills in a specific area and helping you to be more efficient in what you want to improve.

Israel-Laguan replied:
Thanks, @anaceciliamolinares so that we will use WORKSHOP

e-azocar commented on src/actions/EventsActions.ts:
```diff
@@ -0,0 +1,25 @@
+
+export const getListOfEventsForHome = (
+  variables: GetListOfEventsForHomeProps = {
+    limit: 4,
```
Could you change this *Magic Number* for a constant?
For instance:
```javascript
const HOME_EVENTS_LIMIT = 4
```

e-azocar commented on src/pages/Home/index.vue:
```diff
@@ -1,11 +1,29 @@
+    const providers = ref<Provider[]>();
+
+    events.value = await getListOfEventsForHome();
+    await getListOfProviders({ limit: 6 }).then(({ items }) => {
```
Same here. Maybe:
```javascript
const HOME_PROVIDERS_LIST = 6
```

heliomar-pena committed:
Comment changes

heliomar-pena committed:
Refactor and cleaning code

Israel-Laguan commented:
@angelik0828 please check
